From bd468ad296ed5bed2e51ad335858e0c92290e492 Mon Sep 17 00:00:00 2001
From: Dan Callaghan <djc@djc.id.au>
Date: Sun, 18 May 2025 14:32:32 +1000
Subject: [PATCH] add missing dummy arguments in some unprototyped functions

In C23, these functions are now considered to have void args rather than
being unprototyped, which would result in type errors. They actually
take (and ignore) mNcurses as their first argument.

Add a VALUE argument named "dummy" for consistency with all the others
which are declared that way.

Closes #40.

Co-authored-by: involution <involution@gmail.com>
---
 ncurses_wrap.c | 34 +++++++++++++++++-----------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/ncurses_wrap.c b/ncurses_wrap.c
index 8f10f54..c3efd9f 100644
--- a/ncurses_wrap.c
+++ b/ncurses_wrap.c
@@ -146,9 +146,9 @@ init_constants_1(void)
 #endif
 }
 
-static VALUE rbncurs_COLORS()
+static VALUE rbncurs_COLORS(VALUE dummy)
 {return INT2NUM(COLORS);}
-static VALUE rbncurs_COLOR_PAIRS()
+static VALUE rbncurs_COLOR_PAIRS(VALUE dummy)
 {return INT2NUM(COLOR_PAIRS);}
 
 static
@@ -371,7 +371,7 @@ init_functions_0(void)
     NCFUNC(wgetnstr, 3);
 }
 
-static VALUE get_stdscr()
+static VALUE get_stdscr(VALUE dummy)
 {
     VALUE rb_stdscr = rb_iv_get(mNcurses, "@stdscr");
     if (rb_stdscr == Qnil) {
@@ -380,7 +380,7 @@ static VALUE get_stdscr()
     }
     return rb_stdscr;
 }
-static VALUE get_curscr()
+static VALUE get_curscr(VALUE dummy)
 {
     VALUE rb_curscr = rb_iv_get(mNcurses, "@curscr");
     if (rb_curscr == Qnil) {
@@ -390,7 +390,7 @@ static VALUE get_curscr()
     return rb_curscr;
 }
 #ifdef HAVE_NEWSCR
-static VALUE get_newscr()
+static VALUE get_newscr(VALUE dummy)
 {
     VALUE rb_newscr = rb_iv_get(mNcurses, "@newscr");
     if (rb_newscr == Qnil) {
@@ -400,15 +400,15 @@ static VALUE get_newscr()
     return rb_newscr;
 }
 #endif
-static VALUE get_LINES()   {return INT2NUM(LINES);}
-static VALUE get_COLS()    {return INT2NUM(COLS);}
+static VALUE get_LINES(VALUE dummy)   {return INT2NUM(LINES);}
+static VALUE get_COLS(VALUE dummy)    {return INT2NUM(COLS);}
 #ifdef HAVE_TABSIZE
-static VALUE get_TABSIZE() {return INT2NUM(TABSIZE);}
+static VALUE get_TABSIZE(VALUE dummy) {return INT2NUM(TABSIZE);}
 #endif
 #ifdef HAVE_ESCDELAY
 /* This global was an undocumented feature under AIX curses. */
 /* ESC expire time in milliseconds                           */
-static VALUE get_ESCDELAY(){return INT2NUM(ESCDELAY);}
+static VALUE get_ESCDELAY(VALUE dummy){return INT2NUM(ESCDELAY);}
 static VALUE set_ESCDELAY(VALUE dummy, VALUE new_delay)
 {
     ESCDELAY=NUM2INT(new_delay);
@@ -419,7 +419,7 @@ static VALUE set_ESCDELAY(VALUE dummy, VALUE new_delay)
 /* This global is wrapper-specific. It denotes the interval after which the
    terminal is periodically checked for having resized or not. */
 /* time in milliseconds                           */
-static VALUE get_RESIZEDELAY(){return rb_iv_get(mNcurses, "@resize_delay");}
+static VALUE get_RESIZEDELAY(VALUE dummy){return rb_iv_get(mNcurses, "@resize_delay");}
 static VALUE set_RESIZEDELAY(VALUE dummy, VALUE rb_new_delay)
 {
     int c_new_delay = NUM2INT(rb_new_delay);
@@ -480,7 +480,7 @@ static VALUE rbncurs_keybound(VALUE dummy, VALUE keycode, VALUE count)
 }
 #endif
 #ifdef HAVE_CURSES_VERSION
-static VALUE rbncurs_curses_version(){return rb_str_new2(curses_version());}
+static VALUE rbncurs_curses_version(VALUE dummy){return rb_str_new2(curses_version());}
 #endif
 #ifdef HAVE_DEFINE_KEY
 static VALUE rbncurs_define_key(VALUE dummy, VALUE definition, VALUE keycode)
@@ -504,7 +504,7 @@ static VALUE rbncurs_resizeterm(VALUE dummy, VALUE lines, VALUE columns)
 }
 #endif
 #ifdef HAVE_USE_DEFAULT_COLORS
-static VALUE rbncurs_use_default_colors()
+static VALUE rbncurs_use_default_colors(VALUE dummy)
 {
     return INT2NUM(use_default_colors());
 }
@@ -849,7 +849,7 @@ static int rbncurshelper_do_wgetch_functor (WINDOW *c_win, wgetch_func _wgetch_f
     double delay = (screen_delay > 0) ? screen_delay : window_delay;
     int result;
     double starttime, nowtime, finishtime;
-    double resize_delay = NUM2INT(get_RESIZEDELAY()) / 1000.0;
+    double resize_delay = NUM2INT(get_RESIZEDELAY(mNcurses)) / 1000.0;
     fd_set in_fds;
 #ifdef HAVE_CLOCK_GETTIME
     struct timespec tv;
